src/Adapter/MSTest.TestAdapter/Execution/TestClassInfo.cs

```diff
@@ -395,14 +395,9 @@ public void RunClassInitialize(TestContext testContext)
                         errorMessage,
                         exceptionStackTraceInfo?.ErrorStackTrace);
 
-                    if (classCleanupLifecycle == ClassCleanupBehavior.EndOfClass)
-                    {
-                        var testFailedException = new TestFailedException(ObjectModelUnitTestOutcome.Failed, errorMessage, exceptionStackTraceInfo);
-                        ClassCleanupException = testFailedException;
-                        throw testFailedException;
```

Comment:
I will need to double check if we don't want to keep the throw behavior to short-circuit some other calls.

Reply:
As far as I can see now the method is unused except from tests. It seems to be related to your previous PR where you changed default behavior (I am investigating the older PR).

Reply:
Seems to be dead code altogether. I think you can remove it all.